### PR TITLE
Proposal to #283

### PR DESCRIPTION
**Pull request by ebaklund**
A bit worried that init_buffer() is not much used.
Would like to have more tests than only --example instancing.
Also, semantics of usage-flags differ significantly between glBufferStorage and glBufferData.
Kind regards,
Erik.

### CODE REVIEW

**Comment by ebaklund** [comment]
I walked the dog and realized that of course, init_buffer() is indeed used - a lot.
Running through the examples I see that my proposal does not work when animations are involved.
So, it might be that glBufferStorage() cannot be as easy an substitute for glBufferData() as I first thought.

**Comment by ebaklund** [comment]
Closing until solution is found.

**Comment by kvark** [comment]
I'll check it out tomorrow and should come up with some ideas.

**Comment by ebaklund** [comment]
Seems the animated examples are broken upstream. Those should be fixed before we go further with glBufferStorage(),

**Comment by ebaklund** [comment]
Migration to std::time broke the animations.

**Comment by ebaklund** [comment]
To be clear: My glBufferStore() proposal seem to work once the std::time related regressions were fixed.

File: src/backend/gl/src/factory.rs [hunk 1]
```diff
@@ -101,19 +101,35 @@ impl Factory {
...
-            gl.BufferData(target,
-                info.size as gl::types::GLsizeiptr,
-                0 as *const gl::types::GLvoid,
-                usage
-            );
+        if self.share.private_caps.buffer_storage_supported {
+            let usage = match info.usage {
+                f::Usage::GpuOnly    => 0,
+                _                    => gl::MAP_WRITE_BIT | gl::DYNAMIC_STORAGE_BIT,
+            };
```
Comment (kvark) [inline src/backend/gl/src/factory.rs]: don't worry about this too much, the usage bits are to be refactored soon

File: src/backend/gl/src/info.rs [hunk 2]
```diff
@@ -248,6 +249,7 @@ pub fn get(gl: &gl::Gl) -> (Info, Capabilities, PrivateCaps) {
         immutable_storage_supported:       info.is_version_or_extension_supported(4, 2, "GL_ARB_texture_storage"),
         sampler_objects_supported:         info.is_version_or_extension_supported(3, 3, "GL_ARB_sampler_objects"),
         program_interface_supported:       info.is_version_or_extension_supported(4, 3, "GL_ARB_program_interface_query"),
+        buffer_storage_supported:          info.is_version_or_extension_supported(4, 4, "GL_ARB_program_interface_query"),
```
Comment (kvark) [inline src/backend/gl/src/info.rs]: wrong extension check (`GL_ARB_program_interface_query`)

**Comment by kvark** [comment]
Thanks @ebaklund ! Please fix the extension name to proceed.

**Comment by kvark** [comment]
@homu r+

**Pull request merged**